src/components/slider.js

```diff
@@ -0,0 +1,371 @@
+      animating: false,
+      animation: undefined
+    });
+    this.setupAutoplay();
+    if (typeof this.props.onSlideChange === "function") {
+      this.props.onSlideChange({ slideIndex: this.nextSlideIndex });
+    }
+  };
+
+  isDisabled = () =>
+    this.slideCount < 2 || this.state.animating || this.props.disabled;
+
+  isInfinite = () => this.slideCount > 2 && this.props.infinite !== false;
+
+  canGoPrevious = () => this.isInfinite() || this.state.currentSlideIndex > 0;
+
+  canGoNext = () =>
+    this.isInfinite() || this.state.currentSlideIndex < this.slideCount - 1;
+
+  goTo = (index, animation) => {
+    if (this.isDisabled()) return;
+    this.nextSlideIndex = index;
+    this.setState({ animating: true, animation });
+    const timeout = this.props.duration || DEFAULT_DURATION;
+    this.animationTimerId = setTimeout(this.onAnimationEnd, timeout);
+  };
+
+  previous = () => {
+    if (!this.canGoPrevious()) return;
+    const nextSlideIndex = this.state.currentSlideIndex - 1;
+    const actualNextSlide =
+      nextSlideIndex >= 0 ? nextSlideIndex : this.slideCount - 1;
+    this.goTo(actualNextSlide, PREVIOUS);
+  };
+
+  next = () => {
+    if (!this.canGoNext()) return;
+    const nextSlideIndex = (this.state.currentSlideIndex + 1) % this.slideCount;
+    this.goTo(nextSlideIndex, NEXT);
+  };
+
+  getSlideClass = index => {
+    const { currentSlideIndex, animation } = this.state;
+    const classNames = this.getClassNames();
+    const lastSlideIndex = this.slideCount - 1;
+    if (index === currentSlideIndex) {
+      if (animation) return `${classNames.animateOut} ${classNames[animation]}`;
+      return classNames.current;
+    }
+    if (this.slideCount === 2) {
+      if (animation) return `${classNames.animateIn} ${classNames[animation]}`;
+      return index < currentSlideIndex ? classNames.previous : classNames.next;
+    }
+    if (
+      index === currentSlideIndex - 1 ||
+      (currentSlideIndex === 0 && index === lastSlideIndex)
+    ) {
+      if (animation === PREVIOUS)
+        return `${classNames.animateIn} ${classNames.previous}`;
+      if (animation === NEXT) return classNames.hidden;
+      return classNames.previous;
+    }
+    if (
+      index === currentSlideIndex + 1 ||
+      (index === 0 && currentSlideIndex === lastSlideIndex)
+    ) {
+      if (animation === NEXT)
+        return `${classNames.animateIn} ${classNames.next}`;
+      if (animation === PREVIOUS) return classNames.hidden;
+      return classNames.next;
+    }
+    return classNames.hidden;
+  };
+
+  /* eslint-disable lines-between-class-members */
+  isSwiping = false;
+  sliderRef;
+  pageStartPosition;
+
+  currentElement;
+  currentElementStartPosition;
+  currentElementPosition;
+  previousElement;
+  previousElementStartPosition;
+  previousElementPosition;
+  nextElement;
+  nextElementStartPosition;
+  nextElementPosition;
+  /* eslint-enable lines-between-class-members */
+
+  handleTouchStart = e => {
```

Comment:
I recommend writing out entirely the variable name,`e` should become `event` just to provide more obvious context.